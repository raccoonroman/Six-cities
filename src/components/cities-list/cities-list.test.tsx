import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import CitiesList from './cities-list';


const mockStore = configureStore([]);

const cities: string[] = [`Kyiv`, `Kharkiv`, `Odessa`, `Vinnytsia`, `Lviv`];
const currentCity = `Vinnytsia`;


it(`Render <CitiesList />`, () => {
  const store = mockStore({
    cities: {
      cities,
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <CitiesList currentCity={currentCity} />
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
