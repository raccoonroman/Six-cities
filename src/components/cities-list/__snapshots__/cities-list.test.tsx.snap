// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <CitiesList /> 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Kyiv
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Kharkiv
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Odessa
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="#"
      onClick={[Function]}
    >
      <span>
        Vinnytsia
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Lviv
      </span>
    </a>
  </li>
</ul>
`;
